File: API/Entities/Item.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class Item:BaseEntity
    {
        public ItemType ItemType { get; set; }
        [Required]
        public string Name { get; set; }
        public string Unit { get; set; }
        public string Currency { get; set; }
        public decimal SellingPrice { get; set; }
        public string  Description { get; set; }
        public  ICollection<InvoiceItem> Invoice { get; set; } = new HashSet<InvoiceItem>();
    }
    public enum ItemType
    {
        Goods,
        Service,
    }
}

File: Client/src/app/_model/item.ts
import { Guid } from 'guid-typescript';
import { Tax } from './tax';

export interface Item {
  id: Guid;
  itemType: ItemType;
  name: string;
  unit: string;
  currency: string;
  sellingPrice: number;
  quantity: number;
  description: string;
}

export enum ItemType {
  Goods,
  Service,
}
